src/components/modals/SurveyModal.tsx

```diff
@@ -71,28 +101,149 @@ export const SurveyModal = (props: { bridge: AirshipBridge<void> }) => {
       })
     )
 
     bridge.resolve()
   })
 
-  const handleDismissButtonPress = useHandler(async () => {
-    dispatch(logEvent('Survey_Discover', { surveyResponse: 'DISMISSED' }))
+  const handleModalDismiss = useHandler(async () => {
+    dispatch(logEvent('Survey_Discover2', { surveyResponse2: 'DISMISSED' }))
 
     bridge.resolve()
   })
 
-  const handleModalDismiss = useHandler(() => {
-    dispatch(logEvent('Survey_Discover', { surveyResponse: 'DISMISSED' }))
+  const animatedStyle = useAnimatedStyle(() => ({
+    height: inputHeight.value,
+    width: '100%',
+    opacity: inputHeight.value ? 1 : 0
+  }))
 
-    bridge.resolve()
+  return (
+    <EdgeModal
+      bridge={bridge}
+      onCancel={handleModalDismiss}
+      title={
+        <View style={styles.titleContainer}>
+          <Paragraph center>
+            <HeaderText>{sprintf(lstrings.survey_discover_title_1s, config.appName)}</HeaderText>
+          </Paragraph>
+          <Paragraph center>
+            <SmallText>{lstrings.survey_discover_subtitle}</SmallText>
+          </Paragraph>
+        </View>
+      }
+    >
+      {/** HACK: iOS and Android use extraScrollHeight differently... */}
+      <KeyboardAwareScrollView
+        extraScrollHeight={Platform.OS === 'ios' ? theme.rem(-16) : theme.rem(9)}
+        enableOnAndroid
+        contentContainerStyle={styles.contentContainerStyle}
+        keyboardShouldPersistTaps="handled"
+        style={styles.containerStyle}
+      >
+        <EdgeCard>
+          <View style={styles.radioContainer}>
+            {options.map((option, index) => (
+              <Radio value={option.selected} onPress={() => handleOptionPress(index)} key={index}>
+                <EdgeText style={styles.radioLabel}>{option.label}</EdgeText>
+              </Radio>
+            ))}
+            <Animated.View style={[styles.baseAnimatedStyle, animatedStyle]}>
+              <SimpleTextInput value={otherText} onChangeText={setOtherText} placeholder={lstrings.specify_placeholder} horizontalRem={0.5} bottomRem={0.5} />
+            </Animated.View>
+          </View>
+        </EdgeCard>
+      </KeyboardAwareScrollView>
+      <ModalButtons
+        primary={{
+          label: isOtherSelected ? lstrings.survey_opt_submit : lstrings.string_next_capitalized,
+          disabled: selectedIndex == null || (isOtherSelected && otherText.trim() === ''),
+          onPress: isOtherSelected && otherText.trim() !== '' ? handleSubmitPress : handleNextPress,
+          spinner: false
+        }}
+        secondary={{
+          label: lstrings.survey_opt_dismiss,
+          onPress: handleModalDismiss,
+          spinner: false
+        }}
+      />
+    </EdgeModal>
+  )
+}
+
+const SurveyModal2 = (props: { bridge: AirshipBridge<boolean>; category: string }) => {
+  const { bridge, category } = props
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const [options, setOptions] = useState([
```

Comment:
You are going to re-caculate this initial value on every render. Very inefficient. Instead, do `useState(() => [...`, putting a function wrapper around the initializer. Then it will only run once at mount.